### PR TITLE
fix(amplify-frontend-javascript): use CommonJS exports to support Node env  + e2e test

### PR DESCRIPTION
**Pull request by bombguy**
#### Description of changes
Adding e2e tests to ensure compatibility across both es6 & commonJS frontend applications with [fix(amplify-frontend-javascript): use CommonJS exports to support Node env #13174](https://github.com/aws-amplify/amplify-cli/pull/13174)

#### Issue #, if available
https://github.com/aws-amplify/amplify-cli/pull/13174

#### Description of how you validated changes
Verified the web apps created from e2e test and saw that they were able to `import` / `require` `aws-exports` file into the app.

#### Checklist
- [X] PR description included
- [X] `yarn test` passes
- [X] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/aws-exports/require_aws_exports.test.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    await nspawn(getNpmPath(), ['run', 'build'], { cwd: appRoot })
+      .wait(/Compiled successfully./)
+      .runAsync();
+  });
+});
+
+async function createReactApp(cwd: string, projectName: string) {
+  const projectRoot = `${cwd}/${projectName}`;
+
+  execa.sync(getNpxPath(), ['create-react-app', projectName, '--use-npm'], { cwd, encoding: 'utf-8' });
```
Comment (sobolk) [inline packages/amplify-e2e-tests/src/__tests__/aws-exports/require_aws_exports.test.ts:116]: Please pin scripts version, see https://github.com/search?q=repo%3Aaws-amplify%2Famplify-cli+scripts-version&type=code .

File: packages/amplify-e2e-tests/src/__tests__/aws-exports/require_aws_exports.test.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    // check if amplify init generated project files
+    expect(fs.existsSync(`${appRoot}/amplify/.config/project-config.json`)).toBeTruthy();
+    expect(fs.existsSync(`${appRoot}/src/aws-exports.js`)).toBeTruthy();
+    expect(fs.existsSync(`${appRoot}/next.config.js`)).toBeTruthy();
+
+    fs.writeFileSync(`${appRoot}/pages/index.js`, nextIndexContent, { encoding: 'utf8', flag: 'w' });
+    fs.writeFileSync(`${appRoot}/next.config.js`, nextConfigContent, { encoding: 'utf8', flag: 'w' });
+
+    // this will attempt to compile and start next server.
```
Comment (sobolk) [inline packages/amplify-e2e-tests/src/__tests__/aws-exports/require_aws_exports.test.ts:84]: if this is spawning next server process then test worker may hang waiting for child process to finish.
Could you check if this is the case? if yes we should either kill the child process or find other command.
If this is working as expected then it's fine.

**Review by sobolk — COMMENTED** [review]
The change is looking good.

Please make sure to :
- `yarn split-e2e-tests-codebuild` and commit and push the delta.
- `yarn cloud-pr` (to kick off PR build after touching codebuild files)
- `yarn cloud-e2e` to confirm that new test is going to pass in automation.

File: packages/amplify-e2e-tests/src/__tests__/aws-exports/require_aws_exports.test.ts (hunk 2 above)
Comment (bombguy) [reply packages/amplify-e2e-tests/src/__tests__/aws-exports/require_aws_exports.test.ts:84]: This was working as intended when testing in my machine

**Pull request closed without merging**